rcl_action/test/rcl_action/test_action_server.cpp

```diff
@@ -0,0 +1,738 @@
+  rcl_action_server_t action_server = rcl_action_get_zero_initialized_server();
+
+  // Initialize with a null action server
+  ret = rcl_action_server_init(nullptr, &node, ts, action_name, &options);
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Initialize with a null node
+  ret = rcl_action_server_init(&action_server, nullptr, ts, action_name, &options);
+  EXPECT_EQ(ret, RCL_RET_NODE_INVALID) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Initialize with an invalid node
+  rcl_node_t invalid_node = rcl_get_zero_initialized_node();
+  ret = rcl_action_server_init(&action_server, &invalid_node, ts, action_name, &options);
+  EXPECT_EQ(ret, RCL_RET_NODE_INVALID) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Initialize with a null typesupport
+  ret = rcl_action_server_init(&action_server, &node, nullptr, action_name, &options);
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Initialize with a null name
+  ret = rcl_action_server_init(&action_server, &node, ts, nullptr, &options);
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Initialize with a null options
+  ret = rcl_action_server_init(&action_server, &node, ts, action_name, nullptr);
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Initialize with valid arguments
+  ret = rcl_action_server_init(&action_server, &node, ts, action_name, &options);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  // Try to initialize again
+  ret = rcl_action_server_init(&action_server, &node, ts, action_name, &options);
+  EXPECT_EQ(ret, RCL_RET_ALREADY_INIT) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Finalize with null action server
+  ret = rcl_action_server_fini(nullptr, &node);
+  EXPECT_EQ(ret, RCL_RET_ACTION_SERVER_INVALID) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Finalize with invalid action server
+  rcl_action_server_t invalid_action_server = rcl_action_get_zero_initialized_server();
+  ret = rcl_action_server_fini(&invalid_action_server, &node);
+  // Nothing happens
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  // Finalize with null node
+  ret = rcl_action_server_fini(&action_server, nullptr);
+  EXPECT_EQ(ret, RCL_RET_NODE_INVALID) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Finalize with valid arguments
+  ret = rcl_action_server_fini(&action_server, &node);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  // Finalize node
+  ret = rcl_node_fini(&node);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+
+  ret = rcl_shutdown();
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+}
+
+class TestActionServer : public ::testing::Test
+{
+protected:
+  void SetUp()
```

Comment:
@jacobperron missing trailing `override`.

Reply:
Added in 69c0287